### PR TITLE
Enables switching SSL certificates on QUIC with QUICHE

### PR DESCRIPTION
**Pull request by maskit**
This enables using certificates loaded by QUICMultiCertConfigLoader (not only the default cert but also certs for specific servernames and IP addresses).

This also makes other SNI stuff work on QUIC.

### CODE REVIEW

**Comment by maskit** [comment]
This depends on #9322, and most code change will be gone after merging it.

The most important change on this PR is use of `quiche_conn_new_with_tls`. This requires the both ATS and Quiche use the same version of BoringSSL.

**Comment by maskit** [comment]
This requires BoringSSL if we use Quiche implementation, and I think the autest failure is because of the build environment, which use OpenSSL(quictls).

@bneradt Can we use BoringSSL on the image for Quiche build?

**Comment by maskit** [comment]
[approve ci autest]

**Comment by maskit** [comment]
[approve ci autest]

**Comment by maskit** [comment]
This breaks the original QUIC implementation, and autests for HTTP/3 would be ran only if ATS is built with Quiche.

File: iocore/net/QUICNetVConnection_quiche.cc [hunk 1]
```diff
@@ -632,14 +665,83 @@ QUICNetVConnection::protocol_contains(std::string_view tag) const
...
+bool
+QUICNetVConnection::_isTryingRenegotiation() const
+{
+  // Renegotiation is not allowed on QUIC (TLS 1.3) connections.
+  // If handshake is completed when this function is called, that should be unallowed attempt of renegotiation.
+  if (this->getSSLHandShakeComplete()) {
+    return true;
+  } else {
+    return false;
+  }
```
[the remark this answers is not in the document]
Comment (maskit) [reply iocore/net/QUICNetVConnection_quiche.cc]: Made the change.

**Review by bneradt — APPROVED** [review]

**Pull request merged**